app/src/test/java/org/mozilla/fenix/compose/cfr/CFRPopupFullScreenLayoutTest.kt

```diff
@@ -0,0 +1,100 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.compose.cfr
+
+import android.content.Context
+import android.graphics.PixelFormat
+import android.view.View
+import android.view.ViewManager
+import android.view.WindowManager
+import android.view.WindowManager.LayoutParams
+import androidx.lifecycle.ViewTreeLifecycleOwner
+import androidx.savedstate.ViewTreeSavedStateRegistryOwner
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.slot
+import io.mockk.spyk
+import io.mockk.verify
+import mozilla.components.support.test.robolectric.testContext
+import org.junit.Assert.assertEquals
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mozilla.fenix.helpers.FenixRobolectricTestRunner
+
+@RunWith(FenixRobolectricTestRunner::class)
+class CFRPopupFullScreenLayoutTest {
+    @Test
+    fun `WHEN the popup is constructed THEN setup lifecycle owners`() {
```

Comment:
I have been playing around with the tests, and have noticed that both `ViewTreeSavedStateRegistryOwner.get()` and `ViewTreeLifecycleOwner.get()` always return `null`.

Since they are static, do we have any options but to check if the have been called with proper arguments?
Something like this.

```
    @Test
    fun `WHEN the popup is constructed THEN setup lifecycle owners`() {
        mockkStatic(ViewTreeLifecycleOwner::class)
        mockkStatic(ViewTreeSavedStateRegistryOwner::class)

        val container = View(testContext)
        val popupView = CFRPopupFullScreenLayout("", container, mockk(), mockk()) {}

        val viewTreeLifecycleCaptor = slot<View>()
        val viewTreeSavedStateCaptor = slot<View>()

        verify { ViewTreeLifecycleOwner.set(capture(viewTreeLifecycleCaptor), any()) }
        verify { ViewTreeSavedStateRegistryOwner.set(capture(viewTreeSavedStateCaptor), any()) }

        assertEquals(viewTreeLifecycleCaptor.captured, popupView)
        assertEquals(viewTreeSavedStateCaptor.captured, popupView)
    }
```

Reply:
Thank you!
Agree that checking for `null` equality is not really useful.
Your proposed approach would work in checking that whatever value the above "..owner"s have are corrently set, even if they are null.
For a somewhat simpler approach and actually checking for a "real" instance though we can set "..owner"s for the anchor and then test that they are properly propagated to our popupView (after which they will be propagated to `androidx.compose.ui.window.PopupLayout`).